Same here @mapellidario :

```
======================================================================
ERROR: Test the listReplicationRules function, which fetches the container
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/MicroService_t/Tools_t/PycurlRucio_t.py", line 70, in testListReplicationRules
    self.rucioToken, self.tokenValidity = getRucioToken(self.rucioAuthUrl, self.rucioAccount)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/MicroService/Tools/PycurlRucio.py", line 78, in getRucioToken
    res = mgr.getheader(url, params=params, headers=headers, ckey=ckey(), cert=cert())
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Services/pycurl_manager.py", line 324, in getheader
    header, _ = self.request(url, params, headers, verb,
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/Utils/PortForward.py", line 69, in portMangle
    return callFunc(callObj, url, *args, **kwargs)
  File "/home/dmwm/wmcore_unittest/WMCore/src/python/WMCore/Services/pycurl_manager.py", line 306, in request
    raise exc
http.client.HTTPException: url=https://cmsrucio-auth-int.cern.ch/auth/x509, code=401, reason=Unauthorized, headers={'Date': 'Fri, 04 Jun 2021 16:07:57 GMT', 'Server': 'Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_wsgi/4.7.1 Python/3.6', 'Access-Control-Allow-Origin': 'None', 'Access-Control-Allow-Headers': 'None', 'Access-Control-Allow-Methods': '*', 'Access-Control-Allow-Credentials': 'true', 'Access-Control-Expose-Headers': 'X-Rucio-Auth-Token', 'Cache-Control': 'no-cache, no-store, max-age=0, must-revalidate, post-check=0, pre-check=0', 'Pragma': 'no-cache', 'ExceptionClass': 'CannotAuthenticate', 'ExceptionMessage': 'Cannot get DN', 'X-Rucio-Host': 'cmsrucio-auth-int.cern.ch', 'Transfer-Encoding': 'chunked', 'Content-Type': 'application/octet-stream'}
-------------------- >> begin captured logging << --------------------
root: INFO: Requesting a token to Rucio for account: wma_test, against url: https://cmsrucio-auth-int.cern.ch
--------------------- >> end captured logging << ---------------------

```

--- a/src/python/WMCore/Algorithms/ParseXMLFile.py
+++ b/src/python/WMCore/Algorithms/ParseXMLFile.py
@@ -61,7 +61,7 @@ def xmlFileToNode(reportFile):
 
     """
     node = Node("JobReports", {})
-    expat_parse(open(reportFile, 'r'),
+    expat_parse(open(reportFile, 'rb'),
                 build(node))
     return node
 
@@ -76,7 +76,9 @@ def expat_parse(f, target):
     parser = xml.parsers.expat.ParserCreate()
     #parser.buffer_size = 65536
     parser.buffer_text = True
-    parser.returns_unicode = False
+
+    # a leftover from the py2py3 migration - TO BE REMOVED
+    # parser.returns_unicode = False
     parser.StartElementHandler = \
        lambda name,attrs: target.send(('start',(name,attrs)))
     parser.EndElementHandler = \

--- a/src/python/WMCore/Storage/TrivialFileCatalog.py
+++ b/src/python/WMCore/Storage/TrivialFileCatalog.py
@@ -35,7 +35,7 @@ import os
 import re
 
 from urllib.parse import urlsplit
-from xml.dom.minidom import Element
+from xml.dom.minidom import Document
 
 from WMCore.Algorithms.ParseXMLFile import xmlFileToNode
 
@@ -142,15 +142,17 @@ class TrivialFileCatalog(dict):
         """
 
         def _getElementForMappingEntry(entry, mappingStyle):
-            e = Element(mappingStyle)
+            xmlDocTmp = Document()
+            element = xmlDocTmp.createElement(mappingStyle)
             for k, v in viewitems(entry):
                 # ignore empty, None or compiled regexp items into output
                 if not v or (k == "path-match-expr"):
                     continue
-                e.setAttribute(k, str(v))
-            return e
+                element.setAttribute(k, str(v))
+            return element
 
-        root = Element("storage-mapping")  # root element name
+        xmlDoc = Document()
+        root = xmlDoc.createElement("storage-mapping")  # root element name
         for mappingStyle, mappings in viewitems(self):
             for mapping in mappings:
                 mapElem = _getElementForMappingEntry(mapping, mappingStyle)

